### DISCUSSION TITLE
can you provide h function for customize renderer.

### DISCUSSION THREAD

**Discussion by zsytssk** [solidjs/solid#1612] [General]
I'm writing a solidjs customize renderer for weapp support. and I need `h` function in some scenarios. 

I have try use `hyper-dom-expressions.createHyperScript` but I get some errors, weapp environment don't support dom api like `Node`, and customize renderer don't export `assign,dynamicProperty,SVGElements` api.

can you provide a `h` function  or `createHyperScript` for  customize renderer.

**Comment by zsytssk** [comment]
The Customize renderer I am working is not in browser environment, It not have dom api like `Node` and ` document.create` ...
and `[solid/h](https://github.com/solidjs/solid/tree/main/packages/solid/h)` call some `solid-js/web` export api not Customize Renderer api.

**Comment by ryansolid** [comment]
If you don't have a DOM environment then what do you need the `h` function for. Our custom renderer API doesn't have an h function but just gives you hooks to call your own createElement calls. Solid has a `createComponent` call.. so the only real need for the `h` function is to render native elements in the specific platform, so a custom `h` function makes little sense. Like you'd be the one providing it. Have you looked at Solid's universal renderer? I don't quite get the ask I think.
